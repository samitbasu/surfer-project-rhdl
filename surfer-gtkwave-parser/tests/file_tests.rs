use std::path::PathBuf;

fn parse_file(file: &str) {
    let path = PathBuf::from("tests/files").join(file);
    let s = std::fs::read_to_string(path).unwrap();
    let (dirs, errs) = surfer_gtkwave_parser::Parser::new(&s).parse();
    let snapshot = format!(
        "## Input ##\n\n{}\n## Errors ##\n\n{}\n\n## Directives ##\n\n{}",
        s,
        errs.iter()
            .map(|e| format!("{e:?}"))
            .collect::<Vec<_>>()
            .join("\n"),
        dirs.iter()
            .map(|d| format!("{d:?}"))
            .collect::<Vec<_>>()
            .join("\n"),
    );
    let mut settings = insta::Settings::clone_current();
    settings.set_snapshot_suffix(file);
    settings.bind(|| {
        insta::assert_snapshot!(snapshot);
    });
}

#[test]
fn with_8_bit_everything() {
    parse_file("with_8_bit_everything.gtkw");
}

#[test]
fn with_8_bit_one() {
    parse_file("with_8_bit_one.gtkw");
}
