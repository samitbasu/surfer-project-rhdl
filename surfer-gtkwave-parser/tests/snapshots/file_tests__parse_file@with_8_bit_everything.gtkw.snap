---
source: surfer-gtkwave-parser/tests/file_tests.rs
expression: snapshot
---
## Input ##

[*]
[*] GTKWave Analyzer v3.3.118 (w)1999-2023 BSI
[*] Wed Jul 17 07:43:41 2024
[*]
[dumpfile] "/home/gustav/surfer/surfer/examples/with_8_bit.vcd"
[dumpfile_mtime] "Mon Jul  1 13:28:38 2024"
[dumpfile_size] 289
[savefile] "/home/gustav/surfer/surfer/examples/with_8_bit_everything.gtkw"
[timestart] 0
[size] 1280 1346
[pos] -781 -374
*-3.711389 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1
[sst_width] 266
[signals_width] 110
[sst_expanded] 1
[sst_vpaned_height] 410
@22
logic.data[7:0]
@28
logic.data_valid
logic.not_always
[pattern_trace] 1
[pattern_trace] 0

## Errors ##

UnknownDirective("[*]")
UnknownDirective("[*] GTKWave Analyzer v3.3.118 (w)1999-2023 BSI")
UnknownDirective("[*] Wed Jul 17 07:43:41 2024")
UnknownDirective("[*]")
UnknownDirective("[dumpfile_mtime] \"Mon Jul  1 13:28:38 2024\"")
UnknownDirective("[dumpfile_size] 289")
UnknownDirective("[savefile] \"/home/gustav/surfer/surfer/examples/with_8_bit_everything.gtkw\"")
UnknownDirective("[timestart] 0")
UnknownDirective("[size] 1280 1346")
UnknownDirective("[pos] -781 -374")
UnknownLine("*-3.711389 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1 -1")
UnknownDirective("[sst_width] 266")
UnknownDirective("[signals_width] 110")
UnknownDirective("[sst_expanded] 1")
UnknownDirective("[sst_vpaned_height] 410")
UnknownDirective("[pattern_trace] 1")
UnknownDirective("[pattern_trace] 0")

## Directives ##

Dumpfile("/home/gustav/surfer/surfer/examples/with_8_bit.vcd")
Trace { path: "logic.data", bit_range: Some((7, 0)), color: None, flags: Some(Flags(hex | rjustify)) }
Trace { path: "logic.data_valid", bit_range: None, color: None, flags: Some(Flags(bin | rjustify)) }
Trace { path: "logic.not_always", bit_range: None, color: None, flags: None }
